From 8d527f245c58968797c46d5bbb0e8560657ae018 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@linaro.org>
Date: Fri, 13 Nov 2020 13:22:12 +0100
Subject: [PATCH] configs: qemu_tfa_mm32 UEFI StMM in 32bit OP-TEE

qemu_tfa_mm32_defconfig configures a Qemu Armv7-A
emulation where StMM UEFI services are expected from
StMM integration as an OP-TEE secure partition.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
---
 configs/qemu_tfa_mm32_defconfig | 61 +++++++++++++++++++++++++++++++++
 1 file changed, 61 insertions(+)
 create mode 100644 configs/qemu_tfa_mm32_defconfig

diff --git a/configs/qemu_tfa_mm32_defconfig b/configs/qemu_tfa_mm32_defconfig
new file mode 100644
index 0000000000..d82c1d6bb3
--- /dev/null
+++ b/configs/qemu_tfa_mm32_defconfig
@@ -0,0 +1,61 @@
+CONFIG_DEBUG_UART=y
+CONFIG_DEBUG_UART_PL010=y
+CONFIG_DEBUG_UART_BASE=0x09000000
+CONFIG_DEBUG_UART_CLOCK=0
+CONFIG_ARM=y
+CONFIG_ARM_SMCCC=y
+CONFIG_ARCH_QEMU=y
+CONFIG_TFABOOT=y
+CONFIG_ENV_SIZE=0x40000
+CONFIG_ENV_SECT_SIZE=0x40000
+CONFIG_TARGET_QEMU_ARM_32BIT=y
+## CONFIG_ARMV7_LPAE=y  don't include that !!!
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_AHCI=y
+CONFIG_DISTRO_DEFAULTS=y
+CONFIG_FIT=y
+CONFIG_FIT_SIGNATURE=y
+CONFIG_FIT_VERBOSE=y
+CONFIG_FIT_BEST_MATCH=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
+CONFIG_USE_PREBOOT=y
+CONFIG_PREBOOT="pci enum"
+# CONFIG_DISPLAY_CPUINFO is not set
+# CONFIG_DISPLAY_BOARDINFO is not set
+CONFIG_CMD_BOOTEFI_SELFTEST=y
+CONFIG_CMD_NVEDIT_EFI=y
+CONFIG_CMD_PCI=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_EFIDEBUG=y
+CONFIG_OF_BOARD=y
+CONFIG_ENV_IS_IN_FLASH=y
+CONFIG_ENV_ADDR=0x4000000
+CONFIG_SCSI_AHCI=y
+CONFIG_AHCI_PCI=y
+CONFIG_BLK=y
+CONFIG_DM_MMC=y
+CONFIG_SUPPORT_EMMC_RPMB=y
+CONFIG_DM_MTD=y
+CONFIG_MTD_NOR_FLASH=y
+CONFIG_FLASH_CFI_DRIVER=y
+CONFIG_CFI_FLASH=y
+CONFIG_SYS_FLASH_USE_BUFFER_WRITE=y
+CONFIG_SYS_FLASH_CFI=y
+CONFIG_DM_ETH=y
+CONFIG_E1000=y
+CONFIG_NVME=y
+CONFIG_PCI=y
+CONFIG_DM_PCI=y
+CONFIG_PCIE_ECAM_GENERIC=y
+CONFIG_SCSI=y
+CONFIG_DM_SCSI=y
+CONFIG_SYSRESET=y
+CONFIG_SYSRESET_PSCI=y
+CONFIG_TEE=y
+CONFIG_OPTEE=y
+CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_EHCI_PCI=y
+CONFIG_EFI_MM_COMM_TEE=y
+CONFIG_SYS_TEXT_BASE=0x60000000
-- 
2.17.1

